### DISCUSSION TITLE
Proxy server auth mechanism

### DISCUSSION THREAD

**Discussion by shubhrendu** [mockoon/mockoon#505] [Help] [answered]
Is it possible that I can add certificate based or any type of authentication to proxy server?

My proxy server expects authentication...

**Comment by 255kb** [comment]
Hi, 
Mockoon is not doing any type of authentication against the proxied server. However, it forwards everything "as-is". So, If you are sending Authorization headers containing a Basic, Bearer or OAuth authentication, it will go through.

**Reply by shubhrendu** [reply]
Thanks @255kb.

Issue is, my proxy server uses cert based authentication mechanism.

In postman, it asks via popup to select the client certificate while triggering the endpoint.

But in mockoon, it just opens up the basic auth popup, not the popup asking to choose the certificate...

Is there anything, I am missing?

**Reply by 255kb** [reply, marked as answer]
Mockoon does not support any type of cert based authentication and there is no plan to support this in a near future. Feel free to open an issue if you want to propose this new feature.
